File: 백준/Bronze/5073. 삼각형과 세 변/삼각형과 세 변.py
import sys

while True :
    triangle = list(map(int, sys.stdin.readline().split()))

    if sum(triangle) == 0 : break

    triangle.sort()

    if triangle[2] < triangle[1] + triangle[0] :
        if triangle[0] == triangle[2] : print("Equilateral")
        elif triangle[0] == triangle[1] or triangle[1] == triangle[2]: print("Isosceles")
        else : print("Scalene")
    else :
        print("Invalid")

File: 백준/Bronze/5073. 삼각형과 세 변/삼각형과 세 변.java
import java.io.*;
import java.util.*;

public class Main {
	static int n;
	static boolean result;
	static int[][] location;
	static boolean[] v;
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringBuilder sb = new StringBuilder();
		StringTokenizer st;

        while (true) {
			st = new StringTokenizer(br.readLine());
			int x = Integer.parseInt(st.nextToken());	// (t ≤ 50)
			int y = Integer.parseInt(st.nextToken());	// (t ≤ 50)
			int z = Integer.parseInt(st.nextToken());	// (t ≤ 50)
			
			if(x==0 && y==z && z==x) break;
			
			if(x==y && y==z)	 						        sb.append("Equilateral");
			else if(Math.max(Math.max(x, y), z) * 2 >= x+y+z)	sb.append("Invalid");
			else if(x==y || y==z || z==x) 				        sb.append("Isosceles");
			else						 				        sb.append("Scalene");
			   
			sb.append("\n");
		}
		System.out.println(sb);
		br.close();
	}
}